src/parser.y

```diff
@@ -661,15 +671,17 @@ void yyerror (char const *s)
          (*sourcefile()) % lexerget_lineno() % s);
 }
 
-bool parse(FileModule *&module, const char *text, const std::string &filename, int debug)
+bool parse(FileModule *&module, const char *text, const std::string &filename, const std::string &pMainFile, int debug)
 {
   fs::path path = fs::absolute(fs::path(filename));
   
+  mainFile =  std::make_shared<fs::path>(pMainFile);
```

Comment:
Since we always set mainFile, does it really need to be a shared_ptr. Can't we just copy the fs::path object?

Reply:
Oh - yes off course. I used the same data type as parser_sourcefile, but offcourse parser_sourcefile is used a lot more and in different contexts. I mean: The same mainFile path object could be used multiple times - which would make sense - but that would require future changes.
A straigth fs:path object makes more sense in the mean time.